src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1471,7 +1410,7 @@ private static unsafe nuint NarrowUtf16ToAscii_Intrinsified(char* pUtf16Buffer,
             // Turn the 8 ASCII chars we just read into 8 ASCII bytes, then copy it to the destination.
 
             ref byte asciiBuffer = ref *pAsciiBuffer;
-            Vector128<byte> asciiVector = ExtractAsciiVector(utf16VectorFirst, utf16VectorFirst);
+            Vector128<byte> asciiVector = Vector128.Narrow(utf16VectorFirst, utf16VectorFirst);
```

Comment:
@adamsitnik Narrow doesn't know that `utf16VectorFirst` is already ASCII at this point (from my understanding) so it applies a mask via AND to cut anything above 0xFF. Consider this:
![image](https://user-images.githubusercontent.com/523221/181841773-036f6d84-262a-48da-bd16-e395ed083c22.png)

So for this case for better perf you probably want to keep `ExtractAsciiVector`

Reply:
Can we log an issue here for `Narrow`. There is possibly some codegen improvements we can make in .NET 8